src/fuse_concat.cpp

```diff
@@ -98,6 +104,54 @@ struct fused_concat
 MIGRAPHX_REGISTER_OP(fused_concat);
 
 namespace {
+struct find_concat_pointwise
+{
+    auto matcher() const
+    {
+        return match::name("concat")(
+            match::used_once(),
+            match::any_of[match::inputs()](match::name("pointwise")(match::used_once())));
+    }
+
+    void apply(module_pass_manager& mpm, const match::matcher_result& r) const
+    {
+        auto concat_ins = r.result;
+
+        std::vector<instruction_ref> inputs;
+        for(auto input : concat_ins->inputs())
+        {
+            if(input->name() == "pointwise" and input->outputs().size() == 1)
+                inputs.insert(inputs.end(), input->inputs().begin(), input->inputs().end());
+            else
+                inputs.push_back(input);
+        }
+        std::vector<module_ref> module_inputs;
+        std::transform(concat_ins->inputs().begin(),
+                       concat_ins->inputs().end(),
+                       std::back_inserter(module_inputs),
+                       [&](instruction_ref input) {
+                           if(input->name() == "pointwise" and input->outputs().size() == 1)
+                           {
+                               auto* pm = input->module_inputs().front();
+                               return mpm.create_module("concat:" + pm->name(), *pm);
+                           }
+                           auto* pm = mpm.create_module("concat:noop" +
+                                                        std::to_string(get_noop_counter()));
+                           auto x   = pm->add_parameter("x0", shape{input->get_shape().type()});
+                           pm->add_return({x});
+                           return pm;
+                       });
+        auto* post_pm = mpm.create_module("noop:concat" + std::to_string(get_noop_counter()));
+        auto x        = post_pm->add_parameter("!x0", shape{concat_ins->get_shape().type()});
```

Comment:
Why is this prefixed with a `!`? I would prefer not to have it start with a special character because this will become `_x0` in the C++.

Reply:
I wasn't sure. I copied same logic from `pointwise_concat_pointwise` fusion.  I can remove it.

Reply:
It uses `x0` in that pass.

Reply:
https://github.com/ROCm/AMDMIGraphX/blob/b1a24e5420ba49a08f8530cbb2e66264ae145d29/src/fuse_concat.cpp#L158

I am referring this line.